dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/AliasAnalysis.java

```diff
@@ -34,6 +33,24 @@ public interface AliasAnalysis {
 
     boolean mayAlias(MemoryCoreEvent a, MemoryCoreEvent b);
 
+    boolean mustAlias(Alloc a, MemoryCoreEvent e);
+
+    boolean mayAlias(Alloc a, MemoryCoreEvent e);
+
+    boolean mustAlias(Alloc a, MemFree f);
+
+    boolean mayAlias(Alloc a, MemFree f);
+
+    boolean mustAlias(MemFree a, MemFree b);
+
+    boolean mayAlias(MemFree a, MemFree b);
```

Comment:
I know this is a draft, but you don't really want to go with so many overloads, do you?